sdk/core/core-auth/src/tokenCredential.ts

```diff
@@ -49,5 +49,5 @@ export interface AccessToken {
  * @param credential The assumed TokenCredential to be tested.
  */
 export function isTokenCredential(credential: any): credential is TokenCredential {
-  return credential && typeof credential.getToken === "function";
+  return credential && typeof credential.getToken === "function" && credential.getToken.length > 0;
```

Comment:
I can't say that this is bad per se, but in my experience function length checks come back to haunt. Preserving function length during transpilation is... hit or miss. Parameters with defaults don't contribute to length either, so making getToken's parameter optional would make it pass this check. If there's no other way this seems fine but let's be careful!

Reply:
Where is it exactly that we want to differentiate between the ms-rest-nodeauth credential and the TokenCredential?
I wonder, instead of checking for `isTokenCredential()`, you can check the flip side of it.

The `TokenClientCredentials` has 2 functions `signRequest` and `getToken` where as `TokenCredential` has only one of these

Reply:
Yep, that's what I was planning to try after @bterlson's comment.  Just updated the PR with that fix and a test to verify it.